activerecord/test/cases/adapter_test.rb

```diff
@@ -651,7 +651,65 @@ def teardown
         assert_operator Post.count, :>, 0
       end
 
+      test "can reconnect and retry queries under timeout limit when retries are enabled" do
+        skip if @connection.class::ADAPTER_NAME == "SQLite"
+        @connection.stub(:retries_available, 1) do
+          kill_query_process(0.1, /\ASELECT SLEEP\(1\)\z/) do
+            @connection.execute("SELECT SLEEP(1)")
+          end
+        end
+        assert_predicate @connection, :active?
+      end
+
+      test "does not reconnect and retry queries when retries are not enabled" do
+        skip if @connection.class::ADAPTER_NAME == "SQLite"
+        @connection.stub(:connection_retry_timeout, 0) do
+          kill_query_process(0.1, /\ASELECT SLEEP\(1\)\z/) do
+            assert_raises(ActiveRecord::ConnectionFailed) do
+              @connection.execute("SELECT SLEEP(1)")
+            end
+          end
+        end
+        assert_not_predicate @connection, :active?
+      end
+
+      test "does not reconnect and retry queries that exceed timeout limit" do
```

Comment:
Looks like postgres isn't a fan of SLEEP - maybe we can do a normal sleep without a select? Optionally we can test mysql in the mysql adapter test like this and test the postgres adapter in it's own file with postgresql specific calls.

Reply:
Yeah I think we can probably avoid several complications at once by directly testing `with_raw_connection` -- that way we can force retryability, plus we control the block so we can just sleep + raise without needing any DB-specific handling.

(It's private, but it's "public" to the various [internal and external] adapters, so seems fair enough to test it directly.)

Reply:
Okay, perfect ❤️ Was trying to avoid going through any private methods, but this will definitely simplify things!